package junit;

import java.io.File;
import java.util.List;
import java.util.Set;
import org.easymock.EasyMock;
import org.junit.*;
import static org.junit.Assert.*;
import com.syntacticsugar.vooga.util.filefinder.Directory;

/**
 * The class <code>DirectoryTest</code> contains tests for the class <code>{@link Directory}</code>.
 *
 * @generatedBy CodePro at 11/28/15 12:56 AM
 * @author Henry
 * @version $Revision: 1.0 $
 */
public class DirectoryTest {
	/**
	 * Run the Directory(String) constructor test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 11/28/15 12:56 AM
	 */
	@Test
	public void testDirectory_1()
		throws Exception {
		String path = "";

		Directory result = new Directory(path);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at com.syntacticsugar.vooga.util.filefinder.Directory.<init>(Directory.java:40)
		//       at com.syntacticsugar.vooga.util.filefinder.Directory.<init>(Directory.java:20)
		assertNotNull(result);
	}

	/**
	 * Run the Directory(String,int) constructor test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 11/28/15 12:56 AM
	 */
	@Test
	public void testDirectory_2()
		throws Exception {
		String path = "";
		int sort = -1;

		Directory result = new Directory(path, sort);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at com.syntacticsugar.vooga.util.filefinder.Directory.<init>(Directory.java:40)
		assertNotNull(result);
	}

	/**
	 * Run the Directory(String,int) constructor test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 11/28/15 12:56 AM
	 */
	@Test
	public void testDirectory_3()
		throws Exception {
		String path = "";
		int sort = -1;

		Directory result = new Directory(path, sort);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at com.syntacticsugar.vooga.util.filefinder.Directory.<init>(Directory.java:40)
		assertNotNull(result);
	}

	/**
	 * Run the Directory(String,int) constructor test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 11/28/15 12:56 AM
	 */
	@Test
	public void testDirectory_4()
		throws Exception {
		String path = "";
		int sort = -1;

		Directory result = new Directory(path, sort);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at com.syntacticsugar.vooga.util.filefinder.Directory.<init>(Directory.java:40)
		assertNotNull(result);
	}

	/**
	 * Run the Directory(String,int) constructor test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 11/28/15 12:56 AM
	 */
	@Test
	public void testDirectory_5()
		throws Exception {
		String path = "";
		int sort = -1;

		Directory result = new Directory(path, sort);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at com.syntacticsugar.vooga.util.filefinder.Directory.<init>(Directory.java:40)
		assertNotNull(result);
	}

	/**
	 * Run the Directory(String,int) constructor test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 11/28/15 12:56 AM
	 */
	@Test
	public void testDirectory_6()
		throws Exception {
		String path = "";
		int sort = 1;

		Directory result = new Directory(path, sort);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at com.syntacticsugar.vooga.util.filefinder.Directory.<init>(Directory.java:40)
		assertNotNull(result);
	}

	/**
	 * Run the Directory(String,int) constructor test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 11/28/15 12:56 AM
	 */
	@Test
	public void testDirectory_7()
		throws Exception {
		String path = "";
		int sort = 1;

		Directory result = new Directory(path, sort);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at com.syntacticsugar.vooga.util.filefinder.Directory.<init>(Directory.java:40)
		assertNotNull(result);
	}

	/**
	 * Run the Directory(String,int) constructor test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 11/28/15 12:56 AM
	 */
	@Test
	public void testDirectory_8()
		throws Exception {
		String path = "";
		int sort = 1;

		Directory result = new Directory(path, sort);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at com.syntacticsugar.vooga.util.filefinder.Directory.<init>(Directory.java:40)
		assertNotNull(result);
	}

	/**
	 * Run the Directory(String,int) constructor test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 11/28/15 12:56 AM
	 */
	@Test
	public void testDirectory_9()
		throws Exception {
		String path = "";
		int sort = 1;

		Directory result = new Directory(path, sort);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at com.syntacticsugar.vooga.util.filefinder.Directory.<init>(Directory.java:40)
		assertNotNull(result);
	}

	/**
	 * Run the Directory(String,int) constructor test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 11/28/15 12:56 AM
	 */
	@Test
	public void testDirectory_10()
		throws Exception {
		String path = "";
		int sort = 1;

		Directory result = new Directory(path, sort);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at com.syntacticsugar.vooga.util.filefinder.Directory.<init>(Directory.java:40)
		assertNotNull(result);
	}

	/**
	 * Run the Directory(String,int) constructor test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 11/28/15 12:56 AM
	 */
	@Test
	public void testDirectory_11()
		throws Exception {
		String path = "";
		int sort = 1;

		Directory result = new Directory(path, sort);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at com.syntacticsugar.vooga.util.filefinder.Directory.<init>(Directory.java:40)
		assertNotNull(result);
	}

	/**
	 * Run the Directory(String,int) constructor test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 11/28/15 12:56 AM
	 */
	@Test
	public void testDirectory_12()
		throws Exception {
		String path = "";
		int sort = 1;

		Directory result = new Directory(path, sort);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at com.syntacticsugar.vooga.util.filefinder.Directory.<init>(Directory.java:40)
		assertNotNull(result);
	}

	/**
	 * Run the Directory(String,int) constructor test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 11/28/15 12:56 AM
	 */
	@Test
	public void testDirectory_13()
		throws Exception {
		String path = "";
		int sort = 1;

		Directory result = new Directory(path, sort);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at com.syntacticsugar.vooga.util.filefinder.Directory.<init>(Directory.java:40)
		assertNotNull(result);
	}

	/**
	 * Run the Directory(String,int) constructor test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 11/28/15 12:56 AM
	 */
	@Test
	public void testDirectory_14()
		throws Exception {
		String path = "";
		int sort = 1;

		Directory result = new Directory(path, sort);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at com.syntacticsugar.vooga.util.filefinder.Directory.<init>(Directory.java:40)
		assertNotNull(result);
	}

	/**
	 * Run the Directory(String,int) constructor test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 11/28/15 12:56 AM
	 */
	@Test
	public void testDirectory_15()
		throws Exception {
		String path = "";
		int sort = 1;

		Directory result = new Directory(path, sort);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at com.syntacticsugar.vooga.util.filefinder.Directory.<init>(Directory.java:40)
		assertNotNull(result);
	}

	/**
	 * Run the Directory(String,int) constructor test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 11/28/15 12:56 AM
	 */
	@Test
	public void testDirectory_16()
		throws Exception {
		String path = "";
		int sort = 1;

		Directory result = new Directory(path, sort);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at com.syntacticsugar.vooga.util.filefinder.Directory.<init>(Directory.java:40)
		assertNotNull(result);
	}

	/**
	 * Run the Directory(String,int) constructor test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 11/28/15 12:56 AM
	 */
	@Test
	public void testDirectory_17()
		throws Exception {
		String path = "";
		int sort = 1;

		Directory result = new Directory(path, sort);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at com.syntacticsugar.vooga.util.filefinder.Directory.<init>(Directory.java:40)
		assertNotNull(result);
	}

	/**
	 * Run the List<Directory> getDirectories() method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 11/28/15 12:56 AM
	 */
	@Test
	public void testGetDirectories_1()
		throws Exception {
		Directory fixture = new Directory("", 1);

		List<Directory> result = fixture.getDirectories();

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at com.syntacticsugar.vooga.util.filefinder.Directory.<init>(Directory.java:40)
		assertNotNull(result);
	}

	/**
	 * Run the File getFileByName(String) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 11/28/15 12:56 AM
	 */
	@Test
	public void testGetFileByName_1()
		throws Exception {
		Directory fixture = new Directory("", 1);
		String str = "";

		File result = fixture.getFileByName(str);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at com.syntacticsugar.vooga.util.filefinder.Directory.<init>(Directory.java:40)
		assertNotNull(result);
	}

	/**
	 * Run the File getFileByName(String) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 11/28/15 12:56 AM
	 */
	@Test
	public void testGetFileByName_2()
		throws Exception {
		Directory fixture = new Directory("", 1);
		String str = "";

		File result = fixture.getFileByName(str);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at com.syntacticsugar.vooga.util.filefinder.Directory.<init>(Directory.java:40)
		assertNotNull(result);
	}

	/**
	 * Run the File getFileByName(String) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 11/28/15 12:56 AM
	 */
	@Test
	public void testGetFileByName_3()
		throws Exception {
		Directory fixture = new Directory("", 1);
		String str = "";

		File result = fixture.getFileByName(str);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at com.syntacticsugar.vooga.util.filefinder.Directory.<init>(Directory.java:40)
		assertNotNull(result);
	}

	/**
	 * Run the List<File> getFiles() method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 11/28/15 12:56 AM
	 */
	@Test
	public void testGetFiles_1()
		throws Exception {
		Directory fixture = new Directory("", 1);

		List<File> result = fixture.getFiles();

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at com.syntacticsugar.vooga.util.filefinder.Directory.<init>(Directory.java:40)
		assertNotNull(result);
	}

	/**
	 * Run the List<File> getFilesByContains(String) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 11/28/15 12:56 AM
	 */
	@Test
	public void testGetFilesByContains_1()
		throws Exception {
		Directory fixture = new Directory("", 1);
		String str = "";

		List<File> result = fixture.getFilesByContains(str);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at com.syntacticsugar.vooga.util.filefinder.Directory.<init>(Directory.java:40)
		assertNotNull(result);
	}

	/**
	 * Run the List<File> getFilesByContains(String) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 11/28/15 12:56 AM
	 */
	@Test
	public void testGetFilesByContains_2()
		throws Exception {
		Directory fixture = new Directory("", 1);
		String str = "";

		List<File> result = fixture.getFilesByContains(str);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at com.syntacticsugar.vooga.util.filefinder.Directory.<init>(Directory.java:40)
		assertNotNull(result);
	}

	/**
	 * Run the List<File> getFilesByContains(String) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 11/28/15 12:56 AM
	 */
	@Test
	public void testGetFilesByContains_3()
		throws Exception {
		Directory fixture = new Directory("", 1);
		String str = "";

		List<File> result = fixture.getFilesByContains(str);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at com.syntacticsugar.vooga.util.filefinder.Directory.<init>(Directory.java:40)
		assertNotNull(result);
	}

	/**
	 * Run the List<File> getFilesByExtension(String) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 11/28/15 12:56 AM
	 */
	@Test
	public void testGetFilesByExtension_1()
		throws Exception {
		Directory fixture = new Directory("", 1);
		String ext = "";

		List<File> result = fixture.getFilesByExtension(ext);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at com.syntacticsugar.vooga.util.filefinder.Directory.<init>(Directory.java:40)
		assertNotNull(result);
	}

	/**
	 * Run the List<File> getFilesByExtension(String) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 11/28/15 12:56 AM
	 */
	@Test
	public void testGetFilesByExtension_2()
		throws Exception {
		Directory fixture = new Directory("", 1);
		String ext = "";

		List<File> result = fixture.getFilesByExtension(ext);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at com.syntacticsugar.vooga.util.filefinder.Directory.<init>(Directory.java:40)
		assertNotNull(result);
	}

	/**
	 * Run the List<File> getFilesByExtension(String) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 11/28/15 12:56 AM
	 */
	@Test
	public void testGetFilesByExtension_3()
		throws Exception {
		Directory fixture = new Directory("", 1);
		String ext = "";

		List<File> result = fixture.getFilesByExtension(ext);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at com.syntacticsugar.vooga.util.filefinder.Directory.<init>(Directory.java:40)
		assertNotNull(result);
	}

	/**
	 * Run the List<File> getFilesByExtension(String) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 11/28/15 12:56 AM
	 */
	@Test
	public void testGetFilesByExtension_4()
		throws Exception {
		Directory fixture = new Directory("", 1);
		String ext = "";

		List<File> result = fixture.getFilesByExtension(ext);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at com.syntacticsugar.vooga.util.filefinder.Directory.<init>(Directory.java:40)
		assertNotNull(result);
	}

	/**
	 * Run the List<File> getFilesByExtensions(Set<String>) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 11/28/15 12:56 AM
	 */
	@Test
	public void testGetFilesByExtensions_1()
		throws Exception {
		Directory fixture = new Directory("", 1);
		Set<String> ext = EasyMock.createMock(Set.class);
		// add mock object expectations here

		EasyMock.replay(ext);

		List<File> result = fixture.getFilesByExtensions(ext);

		// add additional test code here
		EasyMock.verify(ext);
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at com.syntacticsugar.vooga.util.filefinder.Directory.<init>(Directory.java:40)
		assertNotNull(result);
	}

	/**
	 * Run the List<File> getFilesByExtensions(Set<String>) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 11/28/15 12:56 AM
	 */
	@Test
	public void testGetFilesByExtensions_2()
		throws Exception {
		Directory fixture = new Directory("", 1);
		Set<String> ext = EasyMock.createMock(Set.class);
		// add mock object expectations here

		EasyMock.replay(ext);

		List<File> result = fixture.getFilesByExtensions(ext);

		// add additional test code here
		EasyMock.verify(ext);
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at com.syntacticsugar.vooga.util.filefinder.Directory.<init>(Directory.java:40)
		assertNotNull(result);
	}

	/**
	 * Run the List<File> getFilesByExtensions(Set<String>) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 11/28/15 12:56 AM
	 */
	@Test
	public void testGetFilesByExtensions_3()
		throws Exception {
		Directory fixture = new Directory("", 1);
		Set<String> ext = EasyMock.createMock(Set.class);
		// add mock object expectations here

		EasyMock.replay(ext);

		List<File> result = fixture.getFilesByExtensions(ext);

		// add additional test code here
		EasyMock.verify(ext);
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at com.syntacticsugar.vooga.util.filefinder.Directory.<init>(Directory.java:40)
		assertNotNull(result);
	}

	/**
	 * Run the List<File> getFilesByExtensions(Set<String>) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 11/28/15 12:56 AM
	 */
	@Test
	public void testGetFilesByExtensions_4()
		throws Exception {
		Directory fixture = new Directory("", 1);
		Set<String> ext = EasyMock.createMock(Set.class);
		// add mock object expectations here

		EasyMock.replay(ext);

		List<File> result = fixture.getFilesByExtensions(ext);

		// add additional test code here
		EasyMock.verify(ext);
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at com.syntacticsugar.vooga.util.filefinder.Directory.<init>(Directory.java:40)
		assertNotNull(result);
	}

	/**
	 * Run the boolean hasDirectories() method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 11/28/15 12:56 AM
	 */
	@Test
	public void testHasDirectories_1()
		throws Exception {
		Directory fixture = new Directory("", 1);

		boolean result = fixture.hasDirectories();

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at com.syntacticsugar.vooga.util.filefinder.Directory.<init>(Directory.java:40)
		assertTrue(result);
	}

	/**
	 * Run the boolean hasDirectories() method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 11/28/15 12:56 AM
	 */
	@Test
	public void testHasDirectories_2()
		throws Exception {
		Directory fixture = new Directory("", 1);

		boolean result = fixture.hasDirectories();

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at com.syntacticsugar.vooga.util.filefinder.Directory.<init>(Directory.java:40)
		assertTrue(result);
	}

	/**
	 * Perform pre-test initialization.
	 *
	 * @throws Exception
	 *         if the initialization fails for some reason
	 *
	 * @generatedBy CodePro at 11/28/15 12:56 AM
	 */
	@Before
	public void setUp()
		throws Exception {
		// add additional set up code here
	}

	/**
	 * Perform post-test clean-up.
	 *
	 * @throws Exception
	 *         if the clean-up fails for some reason
	 *
	 * @generatedBy CodePro at 11/28/15 12:56 AM
	 */
	@After
	public void tearDown()
		throws Exception {
		// Add additional tear down code here
	}

	/**
	 * Launch the test.
	 *
	 * @param args the command line arguments
	 *
	 * @generatedBy CodePro at 11/28/15 12:56 AM
	 */
	public static void main(String[] args) {
		new org.junit.runner.JUnitCore().run(DirectoryTest.class);
	}
}